modules: memfault: Bluetooth default metrics (#8329)
nrfconnect/sdk-nrf: samples/bluetooth/peripheral_mds/README.rst

Overview
********

In this sample, the Memfault SDK is used as a module in the |NCS| to collect core dumps, reboot reasons, metrics, and trace events from devices and send them through a Bluetooth gateway to the Memfault cloud.
See `Memfault terminology`_ for more details on the various Memfault concepts.
The sample includes also the BAS functionalities.

Metrics
=======

The sample shows how to capture user-specific metrics.
It defines the following metrics:

   * ``Button3PressCount`` - The number of **Button 3** presses.
   * ``BatteryLvl`` - The simulated battery level.
   * ``Button1TimeMeasure`` - The time measured between two **Button 1** presses.

These metrics are defined in the :file:`samples/bluetooth/peripheral_mds/memfault_config/memfault_metrics_heartbeat_config.def` file.
For more details about the metrics, see `Memfault: Collecting Device Metrics`_.

There are also metrics that are specific to |NCS|.
The :ref:`mod_memfault` adds these system proprietary metrics.
The following metrics are enabled by default in this sample:

   * Bluetooth metrics, enabled and disabled using the :kconfig:option:`CONFIG_MEMFAULT_NCS_BT_METRICS` option.

>       * ``Ncs_BtConnectionCount`` - Bluetooth connection count.
>       * ``Ncs_BtConnectionTime`` - Bluetooth connection time.
>          Amount of time Bluetooth has at least one connection alive.
>       * ``Ncs_BtBondCount`` - Bluetooth bond count.

Reviewer: ```suggestion
      * ``Ncs_BtConnectionCount`` - Number of Bluetooth connections.
      * ``Ncs_BtConnectionTime`` - Bluetooth connection time.
         Amount of time with at least one live Bluetooth connection.
      * ``Ncs_BtBondCount`` - Number of Bluetooth bonds.
```

Author: Done

>    * Stack usage metrics, shows how many bytes of a stack space is free.

Reviewer: ```suggestion
   * Stack usage metrics shows the free stack space in bytes.
```

Author: Done
